### PR TITLE
fix(monday-sdk-js): update GetResponse to reflect the nested data structure

### PR DESCRIPTION
**Pull request by noodny**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://apps.developer.monday.com/docs/advanced-view#saving-persistent-data-using-the-mondaycom-storage-api 
image
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
The implementation hasn't actually changed in a release of the library - from what I understand this is a matter of how the data is sent back from the monday.com APIs, and that's unrelated to the versioning of the SDK.

### CODE REVIEW

**Comment by josh-** [comment]
Thanks @noodny! I wonder if this is a good opportunity to improve the typing for some of the other methods that return objects in a `data` key - perhaps something like this?

```diff
diff --git a/types/monday-sdk-js/index.d.ts b/types/monday-sdk-js/index.d.ts
index 9c2c2204fe..b4cf2228bb 100644
--- a/types/monday-sdk-js/index.d.ts
+++ b/types/monday-sdk-js/index.d.ts
@@ -34,10 +34,8 @@ interface OAuthOptions {
 }
 
 interface GetResponse {
-    data: {
-        value: any;
-        version: any;
-    };
+    value: any;
+    version: any;
 }
 
 interface SetResponse {
@@ -53,7 +51,7 @@ interface MondayClientSdk {
      * Placeholders may be used, which will be substituted by the variables object passed within the options.
      * @param options
      */
-    api(query: string, options?: APIOptions): Promise<any>;
+    api(query: string, options?: APIOptions): Promise<{ data: object }>;
 
     /**
      * Instead of passing the API token to the `api()` method on each request, you can set the API token once using:
@@ -81,7 +79,7 @@ interface MondayClientSdk {
      */
     listen(
         typeOrTypes: SubscribableEvents | ReadonlyArray<SubscribableEvents>,
-        callback: (res: any) => void,
+        callback: (res: { data: object }) => void,
         params?: object,
     ): void;
 
@@ -137,7 +135,7 @@ interface MondayClientSdk {
              */
             excludeCancelButton?: boolean | undefined;
         },
-    ): Promise<any>;
+    ): Promise<{ data: { confirm: boolean } }>;
     /**
      * Display a message at the top of the user's page. Useful for success, error & general messages.
      * @param type Which action to perform
@@ -187,7 +185,7 @@ interface MondayClientSdk {
              * Returns a stored value from the database under `key`
              * @param key
              */
-            getItem(key: string): Promise<GetResponse>;
+            getItem(key: string): Promise<{ data: GetResponse }>;
 
             /**
              * Stores `value` under `key` in the database
@@ -202,7 +200,7 @@ interface MondayClientSdk {
 interface MondayServerSdk {
     setToken(token: string): void;
 
-    api(query: string, options?: Partial<{ token: string, variables: object} >): Promise<any>;
+    api(query: string, options?: Partial<{ token: string; variables: object }>): Promise<any>;
 
     oauthToken(code: string, clientId: string, clientSecret: string): Promise<any>;
 }
diff --git a/types/monday-sdk-js/test/monday-sdk-js-global.test.ts b/types/monday-sdk-js/test/monday-sdk-js-global.test.ts
index 98915c1372..b5a19e1b49 100644
--- a/types/monday-sdk-js/test/monday-sdk-js-global.test.ts
+++ b/types/monday-sdk-js/test/monday-sdk-js-global.test.ts
@@ -1,6 +1,6 @@
 const monday = mondaySdk();
 
-monday.api('test'); // $ExpectType Promise<any>
+monday.api('test'); // $ExpectType Promise<{ data: object; }>
 monday.setToken('test'); // $ExpectType void
 monday.get('context'); // $ExpectType Promise<any>
 monday.get('settings'); // $ExpectType Promise<any>
@@ -8,10 +8,10 @@ monday.get('itemIds'); // $ExpectType Promise<any>
 monday.get('sessionToken'); // $ExpectType Promise<any>
 monday.listen('context', res => res); // $ExpectType void
 monday.execute('openItemCard', { itemId: 123 }); // $ExpectType Promise<any>
-monday.execute('confirm', { message: 'Hello' }); // $ExpectType Promise<any>
+monday.execute('confirm', { message: 'Hello' }); // $ExpectType Promise<{ data: { confirm: boolean; }; }>
 monday.execute('notice', { message: 'Hello' }); // $ExpectType Promise<any>
 monday.oauth({clientId: 'clientId'});
-monday.storage.instance.getItem('test'); // $ExpectType Promise<GetResponse>
+monday.storage.instance.getItem('test'); // $ExpectType Promise<{ data: GetResponse; }>
 monday.storage.instance.setItem('test', '123'); // $ExpectType Promise<SetResponse>
 
 const mondayServer = mondaySdk({ token: '123' });
diff --git a/types/monday-sdk-js/test/monday-sdk-js-module.test.ts b/types/monday-sdk-js/test/monday-sdk-js-module.test.ts
index 43d3148808..284233caea 100644
--- a/types/monday-sdk-js/test/monday-sdk-js-module.test.ts
+++ b/types/monday-sdk-js/test/monday-sdk-js-module.test.ts
@@ -1,7 +1,7 @@
 import mondaySdk from 'monday-sdk-js';
 const monday = mondaySdk();
 
-monday.api('test'); // $ExpectType Promise<any>
+monday.api('test'); // $ExpectType Promise<{ data: object; }>
 monday.setToken('test'); // $ExpectType void
 monday.get('context'); // $ExpectType Promise<any>
 monday.get('settings'); // $ExpectType Promise<any>
@@ -9,10 +9,10 @@ monday.get('itemIds'); // $ExpectType Promise<any>
 monday.get('sessionToken'); // $ExpectType Promise<any>
 monday.listen('context', res => res); // $ExpectType void
 monday.execute('openItemCard', { itemId: 123 }); // $ExpectType Promise<any>
-monday.execute('confirm', { message: 'Hello' }); // $ExpectType Promise<any>
+monday.execute('confirm', { message: 'Hello' }); // $ExpectType Promise<{ data: { confirm: boolean; }; }>
 monday.execute('notice', { message: 'Hello' }); // $ExpectType Promise<any>
 monday.oauth({clientId: 'clientId'});
-monday.storage.instance.getItem('test'); // $ExpectType Promise<GetResponse>
+monday.storage.instance.getItem('test'); // $ExpectType Promise<{ data: GetResponse; }>
 monday.storage.instance.setItem('test', '123'); // $ExpectType Promise<SetResponse>
 
 const mondayServer = mondaySdk({ token: '123' });
```

Since this patch ^ includes changes to the test files it means I'm able to merge this PR without waiting for DT maintainers to approve this PR (as it currently would without changes to the tests) let me know what you think 👍

**Comment by noodny** [comment]
Thanks for the quick reply @josh-! My Monday.com SDK knowledge is limited to what I've used, but I went through the docs and your updates look good to me.

**Comment by josh-** [comment]
Sounds great @noodny, would it be possible for you to add those changes to this PR?

**Review by josh- — APPROVED** [review]
LGTM 🎉

**Comment by josh-** [comment]
Ready to merge

**Pull request merged**
